src/Support/Generators/RelationPropertyGenerator.php

```diff
@@ -52,37 +64,62 @@ public function getPropertyDefinition(Model $model): array
     /**
      * Get return type based on typing or doc block.
      *
-     * @param  ReflectionMethod $method
-     * @return string
+     * @param ReflectionMethod $method
+     * @return array<int, string>
      */
-    private function getReturnType(ReflectionMethod $method): string
+    private function getReturnType(ReflectionMethod $method): array
     {
         if (null !== ($returnType = $method->getReturnType())) {
-            // @phpstan-ignore-next-line
-            return $returnType->getName();
+            if (in_array($returnType::class, self::REFLECTION_RETURN_TYPES)) {
+                /** @phpstan-ignore-next-line: if it gets here the type is definitely correct */
+                return $this->parseReflectionReturnType($returnType);
+            }
         }
 
         $docComment = $method->getDocComment();
 
         if (false === $docComment) {
-            return '';
+            return [''];
```

Comment:
Why are you returning an array with an empty string in all these places now? Is there no better way to solve this?